packages/flutter/lib/src/widgets/text.dart

```diff
@@ -713,3 +719,635 @@ class Text extends StatelessWidget {
     }
   }
 }
+
+class _RichText extends StatelessWidget {
+  const _RichText({
+    this.textKey,
+    required this.text,
+    this.textAlign = TextAlign.start,
+    this.textDirection,
+    this.softWrap = true,
+    this.overflow = TextOverflow.clip,
+    this.textScaler = TextScaler.noScaling,
+    this.maxLines,
+    this.locale,
+    this.strutStyle,
+    this.textWidthBasis = TextWidthBasis.parent,
+    this.textHeightBehavior,
+    this.selectionColor,
+  }) : assert(maxLines == null || maxLines > 0);
+
+  final GlobalKey? textKey;
+  final InlineSpan text;
+  final TextAlign textAlign;
+  final TextDirection? textDirection;
+  final bool softWrap;
+  final TextOverflow overflow;
+  final TextScaler textScaler;
+  final int? maxLines;
+  final Locale? locale;
+  final StrutStyle? strutStyle;
+  final TextWidthBasis textWidthBasis;
+  final ui.TextHeightBehavior? textHeightBehavior;
+  final Color? selectionColor;
+
+  @override
+  Widget build(BuildContext context) {
+    final SelectionRegistrar? registrar = SelectionContainer.maybeOf(context);
+    return RichText(
+      key: textKey,
+      textAlign: textAlign,
+      textDirection: textDirection,
+      locale: locale,
+      softWrap: softWrap,
+      overflow: overflow,
+      textScaler: textScaler,
+      maxLines: maxLines,
+      strutStyle: strutStyle,
+      textWidthBasis: textWidthBasis,
+      textHeightBehavior: textHeightBehavior,
+      selectionRegistrar: registrar,
+      selectionColor: selectionColor,
+      text: text,
+    );
+  }
+}
+
+class _SelectableTextContainer extends StatefulWidget {
```

Comment:
Nit: This makes me think of SelectableText the widget, should it be renamed? No worries if you disagree, "selectable text" is a pretty generic term, it's fine either way.

Reply:
We might want to move this out of here eventually, and put it with `SelectableText`. But I think its fine for now since its private. Something to think about for later might be whether we want `Text` to be completely un-selectable in all cases even when under a `SelectionArea`, and then `SelectableText` will be the text component that is selectable. Having them both selectable, makes me wonder why they both should exist besides historical reasons/breaking changes.

Reply:
Not to start an unrelated naming debate in this comment thread, but I worry about the existing problem where users (especially on the web) use Text widgets expecting them to be selectable when they're not... Not sure exactly how we should organize/name this though. I agree this PR is fine as-is since it's private.

If we do keep both Text and SelectableText though, it would be cool to implement SelectableText simply as a wrapper of Text and SelectionArea.

Reply:
I think that's still the plan for `SelectableText` with https://github.com/flutter/flutter/issues/104547 (Using `Text` and `SelectionArea`). I think at that time we should evaluate whether we should move this `_SelectableTextContainer` to `SelectableText`. I agree though it is a bit confusing for someone that is going to dig into this code.